packages/api/src/record/service/copy.ts

```diff
@@ -0,0 +1,81 @@
+import { getFolderByFolderLinkId } from "../../folder/service";
+import { type FolderRow, FolderStatus } from "../../folder/models";
+import { validateCreateEventRequest } from "../../event/validators";
+import type { CreateEventRequest } from "../../event/models";
+import { getRecordById } from "../service";
+import { type ArchiveRecord, type CopyRecordRequest, RecordStatus, type RecordVO } from "../models";
+import { InternalServerError } from "http-errors";
+
+export const getRecordFromRecordVo = async (
+  recordVO: RecordVO
+): Promise<ArchiveRecord | null> => {
+  // const archiveId = recordData.RecordVO.archiveNbr;
+  let record = null;
+  let recordId = null;
+
+  recordId = { recordId: recordVO.recordId.toString() };
+  [record] = await getRecordById({
+    recordIds: [recordId.recordId],
+    accountEmail: "test11@testt.com", // Where to get this email from?
+  });
+
+  if (record === undefined) {
+    throw new InternalServerError("RecordVO does not exist");
+  }
+  return record;
+};
+
+export const copyRecord = async (
+  recordData: CopyRecordRequest
+): Promise<FolderRow> => {
+  // TODO check space over - How to check if there is enough space?
+  const targetFolder = await getFolderByFolderLinkId({
+    folderLinkId: recordData.FolderDestVO.folder_linkId.toString(),
+  });
+
+  const record = await getRecordFromRecordVo(recordData.RecordVO);
+  if (!record) {
+    throw new InternalServerError("RecordVO does not exist");
+  }
+  // Do we need need to check for read access?
+
+  // Do we need to check for create permission in target folder?
```

Comment:
Yes. The relevant places to check are the `account_archive` entries binding the calling account to the archive(s) where the record and destination folder live, plus any `access` records associated with the record and/or folder

Reply:
Also we should check for publish permissions on the record if it's being copied to a public folder

Reply:
Also checking for write permissions on the destination should include the check that it's not a connector folder.